pkg/ruler/ruler.go

```diff
@@ -967,15 +967,35 @@ func (r *Ruler) Rules(ctx context.Context, in *RulesRequest) (*RulesResponse, er
 		return nil, fmt.Errorf("no user id found in context")
 	}
 
-	groupDescs, err := r.getLocalRules(userID, in.GetFilter())
+	groupDescs, err := r.getLocalRules(userID, *in)
 	if err != nil {
 		return nil, err
 	}
 
 	return &RulesResponse{Groups: groupDescs}, nil
 }
 
-func (r *Ruler) getLocalRules(userID string, ruleTypeFilter RulesRequest_RuleType) ([]*GroupStateDesc, error) {
+type StringFilterSet map[string]struct{}
+
+func MakeStringFilterSet(values []string) StringFilterSet {
```

Comment:
[nit] Do you need to expose this because of GEM? If not, we may keep it unexported.

```suggestion
func makeStringFilterSet(values []string) StringFilterSet {
```

Reply:
No good reason :) I think at the time I thought about moving it to util, but it's so small I left it where it was.